client/client-multi/client-c/src/kaa/kaa_user.c

```diff
@@ -193,6 +213,90 @@ kaa_error_t kaa_user_manager_attach_to_user(kaa_user_manager_t *self
     return KAA_ERR_NONE;
 }
 
+kaa_error_t kaa_user_manager_attach_endpoint(kaa_user_manager_t *self, const char *endpoint_access_token, kaa_endpoint_status_listener_t *listener)
+{
+    KAA_RETURN_IF_NIL2(self, endpoint_access_token, KAA_ERR_BADPARAM);
+
+    KAA_LOG_TRACE(self->logger, KAA_ERR_NONE, "Going to attach to endpoint "
+                                              "(endpoint_access_token id = \"%s\")"
+                                              , endpoint_access_token);
+
+    endpoint_info_t *info = KAA_CALLOC(1, sizeof(endpoint_info_t));
+    if (!info)
+        return KAA_ERR_NOMEM;
+
+    info->endpoint_token_len = strlen(endpoint_access_token);
+    info->endpoint_token     = KAA_MALLOC(info->endpoint_token_len);
+
+    if (!info->endpoint_token) {
+        KAA_FREE(info);
```

Comment:
If you use calloc to allocate memory for the info structure, it 's a better to use a single function-destructor.